src/Bicep.Core/Semantics/Binder.cs

```diff
@@ -22,23 +22,23 @@ public Binder(SyntaxTree syntaxTree, ISymbolContext symbolContext)
             // TODO use lazy or some other pattern for init
             this.syntaxTree = syntaxTree;
             this.TargetScope = SyntaxHelper.GetTargetScope(syntaxTree);
-            var (allDeclarations, outermostScopes) = DeclarationVisitor.GetAllDeclarations(syntaxTree, symbolContext);
-            var uniqueDeclarations = GetUniqueDeclarations(allDeclarations);
-            var builtInNamespacs = GetBuiltInNamespaces(this.TargetScope);
-            this.bindings = GetBindings(syntaxTree, uniqueDeclarations, builtInNamespacs, outermostScopes);
-            this.cyclesBySymbol = GetCyclesBySymbol(syntaxTree, uniqueDeclarations, this.bindings);
+            var (declarations, outermostScopes) = DeclarationVisitor.GetDeclarations(syntaxTree, symbolContext);
```

Comment:
Offline discussion: I found a bunch of places where we say things like *ALL declarations* but really mean *declarations in this scope*. I'm cleaning those up as I find them.